File: android/src/expo/com/smileidentity/expo/views/SmileIDExpoDocumentCaptureView.kt
package com.smileidentity.expo.views

import android.content.Context
import androidx.compose.runtime.Composable
import com.smileidentity.SmileID
import com.smileidentity.shared.RNDocumentCapture
import com.smileidentity.shared.SmileIDViewConfig
import expo.modules.kotlin.AppContext

/**
 * Expo implementation of DocumentCapture using the new architecture
 */
class SmileIDExpoDocumentCaptureView(
    context: Context,
    appContext: AppContext
) : SmileIDExpoView(context, appContext) {

    var countryCode: String? = null
    var documentType: String? = null
    var allowGallerySelection: Boolean = false

    override fun renderContent() {
        setContentWithTheme {
            RenderDocumentCaptureContent()
        }
    }

    @Composable
    private fun RenderDocumentCaptureContent() {
        val config = SmileIDViewConfig(
            userId = userId,
            jobId = jobId,
            allowAgentMode = allowAgentMode ?: false,
            showInstructions = showInstructions,
            skipApiSubmission = skipApiSubmission,
            showAttribution = showAttribution,
            extraPartnerParams = extraPartnerParams,
            countryCode = countryCode,
            documentType = documentType,
            allowGallerySelection = allowGallerySelection
        )

        SmileID.RNDocumentCapture(
            config = config,
            onResult = { result ->
                handleResultCallback(result)
            }
        )
    }
}

File: package/ios/expo/views/SmileIDExpoDocumentCaptureView.swift
import ExpoModulesCore
import UIKit
import SwiftUI
import SmileID

/**
 * Expo implementation of DocumentCapture using the new architecture
 */
class SmileIDExpoDocumentCaptureView: SmileIDExpoView {
    
    var isDocumentFrontSide: Bool = true
    var showConfirmation: Bool = true
    var allowGalleryUpload: Bool = false
    
    override func renderContent() {
        let config = createConfig()
        
        // Use SmileIDExtReactNative to create the view
        let uiView = SmileID.RNDocumentCapture(
            config: config,
            onResult: { [weak self] result in
                self?.handleResultCallback(result)
            }
        )
        
        // Use the base class helper to set the UIView
        setUIView(uiView)
    }
    
    private func createConfig() -> SmileIDViewConfig {
        var config = SmileIDViewConfig()
        config.userId = userId
        config.jobId = jobId
        config.allowAgentMode = allowAgentMode
        config.showInstructions = showInstructions
        config.showAttribution = showAttribution
        config.extraPartnerParams = extraPartnerParams
        config.isDocumentFrontSide = isDocumentFrontSide
        config.showConfirmation = showConfirmation
        config.allowGalleryUpload = allowGalleryUpload
        return config
    }
}